src/Telegram.Bot/Requests/FileRequestBase.cs

```diff
@@ -1,71 +1,45 @@
-using System.Globalization;
-using System.Linq;
+using System.Text;
 
 namespace Telegram.Bot.Requests;
 
-/// <summary>
-/// Represents an API request with a file
-/// </summary>
+/// <summary>Represents an API request with a file</summary>
 /// <typeparam name="TResponse">Type of result expected in result</typeparam>
-
-public abstract class FileRequestBase<TResponse> : RequestBase<TResponse>
+/// <param name="methodName">Bot API method</param>
+public abstract class FileRequestBase<TResponse>(string methodName) : RequestBase<TResponse>(methodName)
 {
-    /// <summary>
-    /// Initializes an instance of request
-    /// </summary>
-    /// <param name="methodName">Bot API method</param>
-    protected FileRequestBase(string methodName)
-        : base(methodName)
-    { }
-
-    /// <summary>
-    /// Initializes an instance of request
-    /// </summary>
-    /// <param name="methodName">Bot API method</param>
-    /// <param name="method">HTTP method to use</param>
-    protected FileRequestBase(string methodName, HttpMethod method)
-        : base(methodName, method)
-    { }
+    readonly static Encoding Latin1 = Encoding.GetEncoding(28591);
 
-    /// <summary>
-    /// Generate multipart form data content
-    /// </summary>
-    /// <param name="fileParameterName"></param>
-    /// <param name="inputFile"></param>
-    /// <returns></returns>
-    protected MultipartFormDataContent ToMultipartFormDataContent(
-        string fileParameterName,
-        InputFileStream inputFile)
+    /// <inheritdoc/>
+    public override HttpContent? ToHttpContent()
     {
-        if (inputFile is null or { Content: null })
+        InputFileConverter.Attachments.Value = null;
+        var utf8Json = JsonSerializer.SerializeToUtf8Bytes(this, GetType(), JsonBotAPI.Options);
+        var attachments = InputFileConverter.Attachments.Value;
+        if (attachments == null)
+            return new ByteArrayContent(utf8Json) { Headers = { ContentType = new("application/json") { CharSet = "utf-8" } } };
+
+        var multipartContent = new MultipartFormDataContent();
+        var firstLevel = JsonDocument.Parse(utf8Json).RootElement.EnumerateObject();
```

Comment:
Why do you need to first serialize and then to deserialize the JSON back and forth?

Reply:
That's actually what the previous STJ code we used did behind the scenes... I analyzed it because I was worried too.
In fact the new approach is even more efficient.